test/stdlib/StringCompatibility.swift

```diff
@@ -356,5 +356,20 @@ Tests.test("LosslessStringConvertible/force unwrap/\(swift)") {
 }
 #endif
 
+#if !swift(>=4)
+Tests.test("popFirst") {
+  var str = "abcdef"
+  expectEqual("a", str.popFirst()!)
```

Comment:
FYI: there is `expectOptionalEqual` family of functions in `StdlibUnittest`. Not big deal, but they will fail properly instead of reporting an unsafe unwrap in case something goes wrong.